src/Qwiq.Core/QueryDefinitionCollection.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Qwiq
+{
+    public class QueryDefinitionCollection : ReadOnlyObjectWithIdCollection<IQueryDefinition, Guid>, IQueryDefinitionCollection
```

Comment:
Does this need to be its own type instead of just `ICollection<T>`?

Reply:
Yes to support indexing on name and id rather than needing to use single/first